frontend/src/api/ApiCaller.tsx

```diff
@@ -455,24 +451,18 @@ export class BackendAPICaller {
         let mission = await this.getMissionRunById(missionId)
 
         if (failedTasksOnly) {
-            mission.tasks = mission.tasks.filter(
-                (task) => task.status !== TaskStatus.PartiallySuccessful && task.status !== TaskStatus.Successful
-            )
-            // Fix task ordering
-            for (let index = 0; index < mission.tasks.length; index++) {
-                mission.tasks[index].taskOrder = index
+            const path = `missions/rerun/${mission.id}`
+            const body = {
+                robotId: mission.robot.id,
             }
+            const result = await BackendAPICaller.POST<unknown, Mission>(path, body).catch((e) => {
+                console.error(`Failed to POST /${path}: ` + e)
+                throw e
+            })
+            return result.content
+        } else {
+            return BackendAPICaller.scheduleMissionDefinition(mission.missionId!, mission.robot.id)
         }
-
-        const customMission = createCustomMission(mission)
```

Comment:
This is the only place that uses this function. The createCustomMission function can also be removed